collect_app/src/androidTest/java/org/odk/collect/android/support/FakeClickableMapFragment.kt

```diff
@@ -22,6 +24,9 @@ class FakeClickableMapFragment : Fragment(), MapFragment {
         readyListener?.onReady(this)
     }
 
+    override val mapFragmentDelegate: MapFragmentDelegate
```

Comment:
To avoid needing to do this, how would you feel about just passing `Settings` (or `SettingsProvider`) into `zoomToCurrentLocation` and grabbing the setting that way. As far as I can see, everywhere we call it has access to `SettingsProvider`.

Reply:
I didn't want to pass any changes to the zoom level directly to the settings, as it could result in multiple consecutive calls (shared prefs save operation is UI blocking). Instead, I decided it would be better to store it in a variable and save it only when the fragment is stopped.

Reply:
I meant that the `zoomToCurrentLocation` implementation would look like something this:

```kotlin
fun zoomToCurrentLocation(center: MapPoint?, settings: Settings) {
    val zoomLevel = if (settings.contains(LAST_KNOWN_ZOOM_LEVEL)) {
        settings.getFloat(LAST_KNOWN_ZOOM_LEVEL)
    } else {
        POINT_ZOOM.toDouble()
    }

    zoomToPoint(
        center,
        zoomLevel.toDouble(),
        true
    )
}
```

That way you don't need to add `mapFragmentDelegate` to the `MapFragment` API.

Reply:
Aha I see but then reading zoomlevel would be different than saving. Saving via fragmentdelegate but reading directly from settings. Plus like above every click on the current location buton (and every automatic zoom) would call settings again and again instead of getting the value from a variable that stores it for a fragment lifetime. I think the current approach is better.